docs/design/2021-09-22-stale-read.md

```diff
@@ -0,0 +1,153 @@
+# Proposal: Support Stale Read in TiDB
+
+- Author(s): [Yisaer](https://github.com/Yisaer) (Song Gao)
+- Last updated: 2021-09-22
+- Tracking Issue: https://github.com/pingcap/tidb/issues/21094
+- Related Document: https://docs.google.com/document/d/1dSbXbudTK-hpz0vIBsmpfuzeQhhTFJ-zDenygt8fmCk/edit?usp=sharing
+
+## Table of Contents
+
+* [Introduction](#introduction)
+* [Motivation or Background](#motivation-or-background)
+* [Detailed Design](#detailed-design)
+    * [Implementation Overview](#implementation-overview)
+    * [New Syntax Overview](#new-syntax-overview)
+* [Compatibility](#compatibility)
+    * [ForBidden Behavior](#forbidden-behavior)
+    * [Statement Priority](#statement-priority)
+
+## Introduction
+
+This proposal aims to support Stale Read and descirbe what Stale Read looks like and how to keep compatibility with other queries.
+
+## Motivation or Background
+
+Currently, TiDB support Follower Read to read data from follower replicas in order to increase the query throughput. TiDB also support Snapshot Read to read old data with given snapshot timestamp. 
+
+However, Follower Read still require an rpc calling between follower replica and leader replica during querying which also increased query latency. 
+
+For Snapshot Read, it only read data from leader replica which might be bottleneck especially in hot spot cases.
+
+Stale Read is more like the combination of Follower Read and Snapshot Read. It supports tidb to read data from both leader replicas and follower replicas with no extra rpc calling during followers and leader. 
+
+It also supports read data with given snapshot timestamp like Snapshot Read.
+
+## Detailed Design
+
+### Implementation Overview
+
+Normally, tidb will send data query request to the storage with given timestamp to the leader replicas. 
+
+In Stale Read, each replica could handle the data query request because every replica maintain the resolved timestamp from leader replica. 
```

Comment:
```suggestion
In Stale Read, each replica could handle the data query request because every replica maintains the resolved timestamp from the leader replica. 
```